FirebaseMessaging/Sources/FIRMessaging.m

```diff
@@ -804,31 +910,36 @@ - (void)unsubscribeFromTopic:(NSString *)topic
                            topic, [FIRMessagingPubSub removePrefixFromTopic:topic]);
   }
   __weak FIRMessaging *weakSelf = self;
-  [self.instanceID instanceIDWithHandler:^(FIRInstanceIDResult *_Nullable result,
-                                           NSError *_Nullable error) {
-    if (error) {
-      FIRMessagingLoggerError(
-          kFIRMessagingMessageCodeMessaging012,
-          @"The unsubscription operation failed due to an error getting the FCM token: %@.", error);
-      if (completion) {
-        completion(error);
-      }
-      return;
-    }
-    FIRMessaging *strongSelf = weakSelf;
-    NSString *normalizeTopic = [[strongSelf class] normalizeTopic:topic];
-    if (normalizeTopic.length) {
-      [strongSelf.pubsub unsubscribeFromTopic:normalizeTopic handler:completion];
-      return;
-    }
-    NSString *failureReason =
-        [NSString stringWithFormat:@"Cannot parse topic name: '%@'. Will not unsubscribe.", topic];
-    FIRMessagingLoggerError(kFIRMessagingMessageCodeMessaging011, @"%@", failureReason);
-    if (completion) {
-      completion([NSError messagingErrorWithCode:kFIRMessagingErrorCodeInvalidTopicName
-                                   failureReason:failureReason]);
-    }
-  }];
+  [self retrieveFCMTokenForSenderID:self.tokenManager.fcmSenderID
+                         completion:^(NSString *_Nullable FCMToken, NSError *_Nullable error) {
+                           if (error) {
+                             FIRMessagingLoggerError(kFIRMessagingMessageCodeMessaging012,
+                                                     @"The unsubscription operation failed due to "
```

Comment:
s/unsubscription/unsubscribe/

Reply:
We will have to adjust integration test with it as well. The same as the subscription message. I can do it in a separate PR so we can merge in g3 to fix integration test as this PR might not be checked in to master for a while.